src/users/users.service.ts

```diff
@@ -23,4 +49,18 @@ export class UsersService {
   remove(id: number) {
     return `This action removes a #${id} user`;
   }
+
+  async getFullUserData(email: string): Promise<User> {
```

Comment:
Will repository methods work here?
https://typeorm.io/#/find-options

Reply:
@tar-aldev 
1) in the DTO I was describe column "password" and "salt" to do not retrieve for each request to DB, for security. 
2) To avoid this for the, we should use "Query Builder"
3) Also we should use Query Builder whenever possible, to improve the performance of working with the database (https://typeorm.io/#/select-query-builder)